ios/FluentUI.Demo/FluentUIDemoTests/HUDTest_SwiftUI.swift

```diff
@@ -8,13 +8,68 @@ import XCTest
 class HUDTestSwiftUI: BaseTest {
     override var controlName: String { "HUD" }
 
+    let sleepSeconds: UInt32 = 5
+
     override func setUpWithError() throws {
         try super.setUpWithError()
         app.staticTexts["SwiftUI Demo"].tap()
+        app.buttons["Present HUD for 3 seconds"].tap()
+        // sleeps to wait for presented HUD to disappear
+        sleep(sleepSeconds)
     }
 
     // launch test that ensures the demo app does not crash and is on the correct control page
     func testLaunch() throws {
         XCTAssertTrue(app.navigationBars.element(matching: NSPredicate(format: "identifier CONTAINS %@", controlName)).exists)
     }
+
+    func testBlocksPresentation() throws {
+        let blocksInteractionSwitch: XCUIElement = app.switches["Blocks interaction"]
+        let presentHUDButton: XCUIElement = app.buttons["Present HUD for 3 seconds"]
+        var blocksInteractionSwitchInitialValue = blocksInteractionSwitch.value
+
+        // switch value should start as 1
+        XCTAssert(blocksInteractionSwitchInitialValue as? String == "1")
+        presentHUDButton.tap()
+        // attempts to interact with "Blocks interaction" switch
+        blocksInteractionSwitch.tap()
+        // switch value should not change (remains 1)
+        XCTAssert(blocksInteractionSwitch.value as? String == blocksInteractionSwitchInitialValue as? String)
+
+        // sleeps to wait for presented HUD to disappear
+        sleep(sleepSeconds)
+
+        // turns off block interaction functionality
+        blocksInteractionSwitch.tap()
+        blocksInteractionSwitchInitialValue = blocksInteractionSwitch.value
```

Comment:
I think it's a little confusing to change what the initial value is - I would only set it once at the top (line 29), if possible change it to a let - on the next line you could check that blocksInteractionSwitch.value != blocksInteractionSwitchInitialValue, and on the last line (51) check that the value has gone back to the initialvalue

Reply:
I added the new asserts but I think it actually makes the code more confusing so I'm going to change back to just the comments.